tests/php/test-class-amp-link-sanitizer.php

```diff
@@ -238,6 +239,50 @@ public function test_amp_to_amp_meta_tag( $sanitizer_args, $expected_meta ) {
 		$this->assertEquals( $expected_meta, $meta_tag->getAttribute( 'content' ) );
 	}
 
+	/**
+	 * Test disabling mobile redirection if the URL is excluded.
+	 */
+	public function test_disable_mobile_redirect_for_excluded_url() {
+		$enable_mobile_redirect = static function () {
+			return [ Option::MOBILE_REDIRECT => true ];
+		};
+
+		add_filter( 'pre_option_' . AMP_Options_Manager::OPTION_NAME, $enable_mobile_redirect );
+
+		$link = home_url( '/' );
+		$dom  = AMP_DOM_Utils::get_dom_from_content( "<a id='link' href='$link'>Foo</a>" );
```

Comment:
I suggest always using the curly brace nation for variable substitution as it is both clearer to read and avoids breaking in some edge cases:
```suggestion
		$dom  = AMP_DOM_Utils::get_dom_from_content( "<a id='link' href='{$link}'>Foo</a>" );
```